src/api/chassis/controller/defaultOdomChassisController.cpp

```diff
@@ -0,0 +1,131 @@
+/**
+ * @author Ryan Benasutti, WPI
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+#include "okapi/api/chassis/controller/defaultOdomChassisController.hpp"
+#include "okapi/api/odometry/odomMath.hpp"
+#include <cmath>
+
+namespace okapi {
+DefaultOdomChassisController::DefaultOdomChassisController(
+  const TimeUtil &itimeUtil,
+  std::unique_ptr<Odometry> iodometry,
+  std::shared_ptr<ChassisController> icontroller,
+  const QLength imoveThreshold,
+  const QAngle iturnThreshold,
+  std::shared_ptr<Logger> ilogger)
+  : OdomChassisController(itimeUtil, std::move(iodometry), imoveThreshold, iturnThreshold),
+    logger(std::move(ilogger)),
+    controller(std::move(icontroller)) {
+}
+
+void DefaultOdomChassisController::driveToPoint(const Point &ipoint,
+                                                const StateMode &imode,
+                                                const bool ibackwards,
+                                                const QLength &ioffset) {
+  auto [length, angle] = OdomMath::computeDistanceAndAngleToPoint(
+    ipoint, odom->getState(StateMode::FRAME_TRANSFORMATION), imode);
+
+  if (ibackwards) {
+    length *= -1;
+    angle += 180_deg;
+  }
+
+  LOG_INFO("DefaultOdomChassisController: Computed length of " +
+           std::to_string(length.convert(meter)) + " meters and angle of " +
+           std::to_string(angle.convert(degree)) + " degrees");
+
+  if (angle.abs() > turnThreshold) {
+    LOG_INFO("DefaultOdomChassisController: Turning " + std::to_string(angle.convert(degree)) +
+             " degrees");
+    controller->turnAngle(angle);
+  }
+
+  if ((length - ioffset).abs() > moveThreshold) {
+    LOG_INFO("DefaultOdomChassisController: Driving " +
+             std::to_string((length - ioffset).convert(meter)) + " meters");
+    controller->moveDistance(length - ioffset);
+  }
+}
+
+void DefaultOdomChassisController::turnToPoint(const Point &ipoint, const StateMode &imode) {
+  const auto angle =
+    OdomMath::computeAngleToPoint(ipoint, odom->getState(StateMode::FRAME_TRANSFORMATION), imode);
+
+  LOG_INFO("DefaultOdomChassisController: Computed angle of " +
+           std::to_string(angle.convert(degree)) + " degrees");
+
+  if (angle.abs() > turnThreshold) {
```

Comment:
I'm not sure if it would be appropriate to have the default behavior be to have a non-zero `turnThreshold` and not move small amounts. I could see users being confused by this behavior, and I've never encountered a situation where I wanted the robot to not move in the case of a tiny adjustment.

Reply:
It makes sense to have a turnThreshold for odometry calculation for filtering purposes, but I don't think that same logic should carry over to this.

Reply:
I will make them zero by default, but they were very important for the autonomous robot. Most of this code started with that bot.

Reply:
I could see where that would be important in the case where trajectories are automatically generated, that makes sense. I like the idea to set their defaults to zero, that seems like a good compromise to me :+1: